### ISSUE TITLE
v4: Fetching all transactions from transaction_pool

### ISSUE THREAD

**Issue by parazyd** [libbitcoin/libbitcoin-server#547]
Hi, is there a way to get an index of all transactions that are currently in the transaction pool?

This functionality is necessary for one method in Electrum (and the Electrum protocol).

Related: https://github.com/parazyd/obelisk/issues/5#issuecomment-836137327

**Comment by evoskuil** [comment]
No, the tx pool is currently uninsexed. The index has been written but not integrated yet.

**Comment by parazyd** [comment]
So it's something you're considering implementing? At least to fetch all txids in the pool?

I suppose that might even be enough, and then fetch_transaction2 could be used while iterating over the ids.

**Comment by evoskuil** [comment]
It doesn’t search history, it collects metadata on incoming txs as they enter the pool. I’ll integrate it once the node is ready.

**Comment by parazyd** [comment]
This is something that would be followed with the transaction subscription interface?

**Comment by evoskuil** [comment]
Unrelated, thx subscription doesn’t require an index.

**Comment by parazyd** [comment]
Ah alright. I misunderstood then.

Thanks a lot! Looking forward to the integration :)

**Comment by evoskuil** [comment]
version 5 core features.
